hedera-node/hedera-file-service-impl/src/main/java/com/hedera/node/app/service/file/impl/handlers/FileGetContentsHandler.java

```diff
@@ -124,15 +139,26 @@ public void validate(@NonNull final QueryContext context) throws PreCheckExcepti
 
     /**
      * Provides file content about a file.
+     *
      * @param fileID the file to get information about
      * @param fileStore the file store
+     * @param config the configuration
      * @return the content about the file
      */
     private @Nullable FileContents contentFile(
-            @NonNull final FileID fileID, @NonNull final ReadableFileStore fileStore) {
+            @NonNull final FileID fileID,
+            @NonNull final ReadableFileStore fileStore,
+            @NonNull final Configuration config) {
         final var meta = fileStore.getFileMetadata(fileID);
         if (meta == null) {
-            return null;
+            if (notGenesisCreation(fileID, config)) {
+                return null;
```

Comment:
Why do we need this ? Will the `fieStore` not contain genesis creations ?

Reply:
We would have committed all genesis changes to state right ?

Reply:
This is to answer queries before handling the genesis transaction.